nexus_constructor/InstrumentView.py

```diff
@@ -0,0 +1,260 @@
+from PySide2.QtWidgets import QWidget, QVBoxLayout
+from PySide2.Qt3DExtras import Qt3DExtras
+from PySide2.Qt3DCore import Qt3DCore
+from PySide2.QtCore import QPropertyAnimation
+from PySide2.QtGui import QVector3D, QColor, QMatrix4x4
+from nexus_constructor.NeutronAnimationController import NeutronAnimationController
+
+
+class InstrumentView(QWidget):
+    """
+    Class for managing the 3D view in the NeXus Constructor. Creates the initial sample, the initial beam, and the
+    neutron animation.
+    :param parent: The MainWindow in which this widget is created. This isn't used for anything but is accepted as an
+                   argument in order to appease Qt Designer.
+    """
+
+    def __init__(self, parent):
+        super().__init__()
+        lay = QVBoxLayout(self)
+        self.view = Qt3DExtras.Qt3DWindow()
+        self.view.defaultFrameGraph().setClearColor(QColor("lightgrey"))
+        container = QWidget.createWindowContainer(self.view)
+        lay.addWidget(container)
+
+        self.view.camera().lens().setPerspectiveProjection(45, 16 / 9, 0.1, 1000)
+        self.view.camera().setPosition(QVector3D(6, 8, 30))
+        self.view.camera().setViewCenter(QVector3D(0, 0, 0))
+
+        self.root_entity = Qt3DCore.QEntity()
+        camera_entity = self.view.camera()
+        cam_controller = Qt3DExtras.QFirstPersonCameraController(self.root_entity)
+        cam_controller.setLinearSpeed(20)
+        cam_controller.setCamera(camera_entity)
+        self.view.setRootEntity(self.root_entity)
+
+        # Initialise materials
+        self.grey_material = Qt3DExtras.QPhongMaterial()
+        self.red_material = Qt3DExtras.QPhongMaterial()
+        self.beam_material = Qt3DExtras.QPhongAlphaMaterial()
+        self.green_material = Qt3DExtras.QPhongMaterial()
+
+        # Initialise cube objects
+        self.sample_cube_dimensions = [1, 1, 1]
+        self.cube_entity = Qt3DCore.QEntity(self.root_entity)
+        self.cube_mesh = Qt3DExtras.QCuboidMesh()
+
+        self.num_neutrons = 9
+
+        # Create lists for neutron-related objects
+        self.neutron_entities = []
+        self.neutron_meshes = []
```

Comment:
let's make this a dictionary, it's not used for anything other than keeping variables in scope.